### PR TITLE
Integrate firestore

### PR DESCRIPTION
**Pull request by nagham27**
Done:
-fixed best before date display for ingredients since it was displaying time
- Add ingredient backend
-Delete ingredient backend
-Fetching ingredients and display in list view from  backend
-Add meal plan backend
-Modify meal plan backend
-Fetching meal plan - this does not connect to UI/list view bc the Adapter implemented is not complete and requires implementation of additional functions [dataObserver stuff].

### CODE REVIEW

**Review by raquibk — COMMENTED** [review]
@nagham27 May I know why the you removed the other constructor in Ingredient.java? If required, please fix the errors caused by removing this (ShoppingListActivity.java, ShoppingListAddItemFragment.java)

File: code/app/src/main/java/com/example/snackoverflow/MainActivity.java [hunk 1]
```diff
@@ -81,46 +87,62 @@ public void onClick(View view) {
...
             if (Objects.equals(meals.get(i).getDate() ,date)){
-                System.out.println("Date is right");
+                System.out.println("Dates are equal");
                 meals.get(i).getMeals().add(recipe);
+                for(Recipe r : meals.get(i).getMeals()) {
+                    System.out.println(r.getTitle());
+
+                }
+                // update recipes
+                FirestoreDatabase.modifyMealPlan(i,meals);
```
Comment (Bhavnoor-Kaur) [inline code/app/src/main/java/com/example/snackoverflow/MainActivity.java]: Can you remove this or comment it out as this function is not yet implemented?

File: code/app/src/main/java/com/example/snackoverflow/MainActivity.java [hunk 2]
```diff
@@ -81,46 +87,62 @@ public void onClick(View view) {
...
         }
         ArrayList<Recipe> recipes = new ArrayList<Recipe>();
         recipes.add(recipe);
-        meals.add(new Mealday(date, recipes));
+        Mealday mealDay = new Mealday(date, recipes);
+        meals.add(mealDay);
         FragmentManager fm = getSupportFragmentManager();
         mealdayAdapter = new MealdayAdapter(this,meals,fm);
         mealslist.setAdapter(mealdayAdapter);
+        FirestoreDatabase.addMealPlan(mealDay);
```
Comment (Bhavnoor-Kaur) [inline code/app/src/main/java/com/example/snackoverflow/MainActivity.java]: Can you also remove this or comment it out as this function is not implemented yet!

**Review by Bhavnoor-Kaur — CHANGES_REQUESTED** [review]
Just added two comments, otherwise looks good to me!

**Review by Bhavnoor-Kaur — APPROVED** [review]

**Pull request merged**